grpcutil/middlewares/gerr/gerr_middleware.go

```diff
@@ -69,9 +69,16 @@ func (m *ErrorMiddleware) handleError(ctx context.Context, handlerErr error) err
 	if asertoErr == nil {
 		asertoErr = errors.ErrUnknown
 	}
+
+	errorLogger := errors.Logger(handlerErr)
```

Comment:
It's a matter of style but I prefer to avoid introducing new local variables into scope unless they're going to be used later on.
```go
if errorLogger := errors.Logger(handlerErr); errorLogger != nil {
    log = errorLogger
}
```

This way `errorLogger` is undefined outside the `if` statement.

Reply:
This logic should also move further up, right after initializing `log`. Otherwise, any logging that takes place earlier in the function (like line 61) would be sent to the wrong logger.